### DISCUSSION TITLE
Bug in Build Project option?

### DISCUSSION THREAD

**Discussion by tmarplatt** [vkbo/novelWriter#1392] [General]
Assuming the "_Replace Unicode in HTML_" Build option replaces all unicode with HTML (hexadecimal?) entities, then it's not currently working, not entirely at least. I've tried entering a few characters and they all remain as Unicode in the generated .htm file.

I've tested with these: `₀₁₂₃₄₅₆₇₈₉₊₋₌₍₎√`

Is this a bug?

**Comment by vkbo** [comment]
No, it's not a bug per say, novelWriter has a dictionary of Unicode characters and their HTML equivalents. It is not a complete list. It only contains the Unicode characters in use by novelWriter. You can see the list here:

https://github.com/vkbo/novelWriter/blob/9fdfe7616ceac11e3d12a622459adf40eefa067f/novelwriter/constants.py#L366

Perhaps the build option in the new build tools should say "Replace common Unicode characters". In any case, the HTML output should be marked as Unicode, and should render just fine with those Unicode characters in it.

I could make it replace all Unicode with the hex equivalents, but it generates a rather messy output, so I decided not to at the time. Maybe that could be a second option.

**Reply by vkbo** [reply]
Response by @tmarplatt moved here:

----

> Maybe that could be a second option.

I was debugging certain issues with my ereader and epub files generated with Calibre. But in the end Unicode was not the issue. As far as it concerns that, this feature is not actually needed.

**Reply by vkbo** [reply]
> I was debugging certain issues with my ereader and epub files generated with Calibre. But in the end Unicode was not the issue. As far as it concerns that, this feature is not actually needed.

It's only needed for generating HTML that is not Unicode, which used to be more common. The point of the feature is to replace Unicode characters that novelWriter inserts. If the writer inserts unicode characters in other ways, that is not covered currently.

I think the best action would be to relabel it and keep the feature. A full Unicode replace may be a bit redundant.
